### PR TITLE
[BENCH-716] AWS Sagemaker notebook connected / connectedPlus tests

### PR DESCRIPTION
**Pull request by dexamundsen**
- AWS Sagemaker notebook connected & connectedPlus tests
- Fix build warning in AWS Sagemaker UT

Connected test (primary ops)
image

ConnectedPlus test (force delete + undo, create undo) 
Note - function name fixed and combined post this run sample
image

### CODE REVIEW

File: service/src/main/java/bio/terra/workspace/common/utils/AwsUtils.java [hunk 1]
```diff
@@ -808,7 +808,6 @@ public static void stopSageMakerNotebook(
     }
   }
 
-  // TODO(BENCH-727): make idempotent
```
Comment (nfoen) [inline service/src/main/java/bio/terra/workspace/common/utils/AwsUtils.java:811]: Is this change related to this PR?

File: service/src/test/java/bio/terra/workspace/common/fixtures/ControlledAwsResourceFixtures.java [hunk 2]
```diff
@@ -206,47 +209,67 @@ public static ControlledAwsS3StorageFolderResource.Builder makeAwsS3StorageFolde
...
           .build(); // wait failure
-  public static final WaiterResponse waiterNotebookException_2 =
-      DefaultWaiterResponse.builder()
+  public static final WaiterResponse<DescribeNotebookInstanceResponse> waiterNotebookException_2 =
+      DefaultWaiterResponse.<DescribeNotebookInstanceResponse>builder()
           .attemptsExecuted(1)
           .exception(AWS_SERVICE_EXCEPTION_2)
           .build(); // wait failure
 
+  public static String getUniqueInstanceName(String resourceName) {
```
Comment (nfoen) [inline service/src/test/java/bio/terra/workspace/common/fixtures/ControlledAwsResourceFixtures.java:228]: If you switch to appendRandNumber for the other PR, we should probably do it here also. It may be too long, but something like "wsm-test-instance-<resource and/or rand-number>" vs "wsm-test-s3-<resource and/or rand-number>" may be helpful

**Review by nfoen — APPROVED** [review]
lgtm

**Review by yuhuyoyo — APPROVED** [review]

File: service/src/main/java/bio/terra/workspace/common/utils/AwsUtils.java (hunk 1 above)
Comment (dexamundsen) [reply service/src/main/java/bio/terra/workspace/common/utils/AwsUtils.java:811]: Not exactly, this was a stale comment. Bench-727 is already fixed

File: service/src/test/java/bio/terra/workspace/common/fixtures/ControlledAwsResourceFixtures.java (hunk 2 above)
Comment (dexamundsen) [reply service/src/test/java/bio/terra/workspace/common/fixtures/ControlledAwsResourceFixtures.java:228]: Considered the length, but since they are only test resources, did not think it would be a concern. 
Note- adding wsm-test as the prefix for resources so that its easy to identify them on the AWS. 
same approach used in CLI AWS tests 
i.e. cli-test-x-resourceName

Comment (dexamundsen) [reply service/src/test/java/bio/terra/workspace/common/fixtures/ControlledAwsResourceFixtures.java:228]: There is another short-random number function I might as well use that

[the remark this answers is not in the document]
Comment (dexamundsen) [reply service/src/test/java/bio/terra/workspace/common/fixtures/ControlledAwsResourceFixtures.java:228]: Changing as per other comment

File: service/src/test/java/bio/terra/workspace/common/fixtures/ControlledAwsResourceFixtures.java [hunk 3]
```diff
@@ -206,47 +209,67 @@ public static ControlledAwsS3StorageFolderResource.Builder makeAwsS3StorageFolde
...
+            WorkspaceFixtures.DEFAULT_USER_EMAIL)
         .build();
   }
 
   public static ControlledAwsSageMakerNotebookResource.Builder
       makeAwsSageMakerNotebookResourceBuilder(String instanceName) {
     return makeAwsSageMakerNotebookResourceBuilder(
-        WORKSPACE_ID, TestUtils.appendRandomNumber("sagemaker-resource"), instanceName);
+        WORKSPACE_ID,
+        UUID.randomUUID().toString(),
```
[the remark this answers is not in the document]
Comment (dexamundsen) [reply service/src/test/java/bio/terra/workspace/common/fixtures/ControlledAwsResourceFixtures.java:255]: Resource names must be unique and using UUID to ensure the uniqueness.

File: service/src/test/java/bio/terra/workspace/service/resource/controlled/cloud/aws/sageMakerNotebook/AwsSageMakerNotebookFlightMiscTest.java [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import software.amazon.awssdk.services.sagemaker.model.NotebookInstanceStatus;
+
+// Misc. connectedPlus tests
+@Tag("aws-connected-plus")
+public class AwsSageMakerNotebookFlightMiscTest extends BaseAwsSageMakerNotebookFlightTest {
+
+  @Test
+  void createSageMakerNotebookUndoTest() throws InterruptedException {
```
[the remark this answers is not in the document]
Comment (dexamundsen) [reply service/src/test/java/bio/terra/workspace/service/resource/controlled/cloud/aws/sageMakerNotebook/AwsSageMakerNotebookFlightMiscTest.java:28]: Sure, no concerns. changing

File: service/src/test/java/bio/terra/workspace/common/fixtures/ControlledAwsResourceFixtures.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (yuhuyoyo) [reply service/src/test/java/bio/terra/workspace/common/fixtures/ControlledAwsResourceFixtures.java:255]: but you created a getUniqueInstanceName method too?

File: service/src/test/java/bio/terra/workspace/service/resource/controlled/cloud/aws/sageMakerNotebook/AwsSageMakerNotebookFlightMiscTest.java [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+import java.util.UUID;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import software.amazon.awssdk.services.sagemaker.model.NotebookInstanceStatus;
+
+// Misc. connectedPlus tests
+@Tag("aws-connected-plus")
+public class AwsSageMakerNotebookFlightMiscTest extends BaseAwsSageMakerNotebookFlightTest {
+
+  @Test
```
[the remark this answers is not in the document]
Comment (dexamundsen) [reply service/src/test/java/bio/terra/workspace/service/resource/controlled/cloud/aws/sageMakerNotebook/AwsSageMakerNotebookFlightMiscTest.java:27]: Yes, it is because of the runtime.
Since 'do' is more critical and is part of regular success flow, it would be run as part of the connectedTest (PR checks, etc). 

Undo flow is relatively rare assuming creation via do goes well (critical nevertheless). Hence made it part of the connectedPlus test which is part of Nightly run. Adding this to the the connected Test will make PR checks runs over an hr easily.

File: service/src/test/java/bio/terra/workspace/common/fixtures/ControlledAwsResourceFixtures.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (dexamundsen) [reply service/src/test/java/bio/terra/workspace/common/fixtures/ControlledAwsResourceFixtures.java:255]: Resource names are used in SAM. instance name is used in AWS. 
The only reason I didnt use a UUID directly for instanceName (and similarly folderName in S3) is to have a prefix (say wsm-test) which can help identify resources created by tests. 
I will update this PR (and S3 tests) to use the same resource / instance name in the format - wsmTest<uuid>, with the appendRandomNumber function.
